src/tools/log_exporter.cc

```diff
@@ -0,0 +1,233 @@
+
+    do {
+        Slice value;
+        status = reader.ReadRecord(&value, &scratch);
+        if (!status.ok()) {
+            // Finish reading the file.
+            break;
+        }
+        ::openmldb::api::LogEntry entry;
+        entry.ParseFromString(value.ToString());
+
+        // Determine if there is a dimension with an idx of 0 in the dimensions.
+        // If so, parse the value, else skip it
+        if (entry.dimensions_size() != 0 && entry.log_index() > offset_) {
+            for (int i = 0; i < entry.dimensions_size(); i++) {
+                if (entry.dimensions(i).idx() == 0) {
+                    std::string row;
+                    row = entry.value();
+                    view.Reset(reinterpret_cast<int8_t*>(&(row[0])), row.size());
+                    WriteToFile(view);
+                    success_cnt++;
+                    break;
+                }
+            }
+        }
+    } while (status.ok());
+    PDLOG(INFO, "success_cnt: %lu", success_cnt);
+    offset_ += success_cnt;
+}
+
+void Exporter::ReadSnapshot() {
+    FILE* fd_r = fopen(snapshot_path_.c_str(), "rb");
+    if (fd_r == NULL) {
+        PDLOG(ERROR, "fopen failed: %s", snapshot_path_.c_str());
+        return;
+    }
+    SequentialFile* rf = NewSeqFile(snapshot_path_, fd_r);
+    std::string scratch;
+    bool is_compress = false;
+    if (snapshot_path_.find(openmldb::log::ZLIB_COMPRESS_SUFFIX) != std::string::npos ||
+        snapshot_path_.find(openmldb::log::SNAPPY_COMPRESS_SUFFIX) != std::string::npos) {
+        is_compress = true;
+    }
+    Reader reader(rf, NULL, true, 0, is_compress);
+    Status status;
+    RowView view(schema_);
+
+    do {
+        Slice value;
+        status = reader.ReadRecord(&value, &scratch);
+        if (!status.ok()) {
+            // Finish reading the file.
+            break;
+        }
+        ::openmldb::api::LogEntry entry;
+        entry.ParseFromString(value.ToString());
+
+        // Determine if there is a dimension with an idx of 0 in the dimensions.
+        // If so, parse the value, else skip it
+        if (entry.dimensions_size() != 0) {
+            for (int i = 0; i < entry.dimensions_size(); i++) {
+                if (entry.dimensions(i).idx() == 0) {
+                    std::string row;
+                    row = entry.value();
+                    view.Reset(reinterpret_cast<int8_t*>(&(row[0])), row.size());
+                    WriteToFile(view);
+                    break;
+                }
+            }
+        }
+    } while (status.ok());
+}
+
+void Exporter::WriteToFile(::openmldb::codec::RowView& view) {
+    // Gets the values for each column, then writes the row to the csv file.
+    for (int i = 0; i< schema_.size() - 1; ++i) {
+        std::string col;
+        view.GetStrValue(i, &col);
+        table_cout_ << col << FLAGS_delimiter;
+    }
+    std::string col;
+    view.GetStrValue(schema_.size() - 1, &col);
+    table_cout_ << col << "\n";
```

Comment:
for (int i = 0; i< schema_.size(); ++i) {
        std::string col;
        view.GetStrValue(i, &col);
        table_cout_ << col；
         if (i < schema_.size() - 1) {
              table_cout_ << FLAGS_delimiter;
         } else {
              table_cout_ << "\n";
         }
  }